geode-gfsh/src/main/java/org/apache/geode/management/internal/cli/commands/ConnectCommand.java

```diff
@@ -186,6 +186,16 @@ public ResultModel connect(
       gfsh.logInfo("failed to get the the remote version.", ex);
     }
 
+    // fallback: see if serialization versions matches (Geode 1.12 or later cluster)
+    try {
+      String gfshGeodeSerializationVersion = gfsh.getGeodeSerializationVersion();
+      String remoteGeodeSerializationVersion = invoker.getRemoteGeodeSerializationVersion();
+      if (gfshGeodeSerializationVersion.equals(remoteGeodeSerializationVersion)) {
+        return result;
+      }
+    } catch (Exception ignore) {
+    }
+
```

Comment:
I've moved the logic for determining compatible clusters into a central method to decouple from the network calls needed to get the necessary info.  It's a lot easier now to test just this method in isolation.

Rather than add even further assumptions about product name (even if it were available from the remote cluster) I've constrained the logic that uses product version to the narrowest possible scenario (specific known versions and only used if the remote cluster does not report serialization version)